armi/materials/tests/test_uZr.py

```diff
@@ -13,16 +13,117 @@
 # limitations under the License.
 
 """Tests for simplified UZr material."""
-import unittest
+from unittest import TestCase
 
-from armi.materials.tests import test_materials
 from armi.materials.uZr import UZr
 
 
-class UZR_TestCase(test_materials._Material_Test, unittest.TestCase):
+class UZR_TestCase(TestCase):
+
     MAT_CLASS = UZr
 
+    def setUp(self):
+        self.mat = self.MAT_CLASS()
+
+    def test_isPicklable(self):
+        """Test that materials are picklable so we can do MPI communication of state.
+
+        .. test:: Test the material base class.
+            :id: T_ARMI_MAT_PROPERTIES0
+            :tests: R_ARMI_MAT_PROPERTIES
+        """
+        stream = pickle.dumps(self.mat)
+        mat = pickle.loads(stream)
+
+        # check a property that is sometimes interpolated.
+        self.assertEqual(
+            self.mat.thermalConductivity(500), mat.thermalConductivity(500)
+        )
+
+    def test_TD(self):
+        """Test the material density.
+
+        .. test:: Test the material base class.
+            :id: T_ARMI_MAT_PROPERTIES2
+            :tests: R_ARMI_MAT_PROPERTIES
+        """
+        self.assertEqual(self.mat.getTD(), self.mat.theoreticalDensityFrac)
+
+        self.mat.clearCache()
+        self.mat._setCache("dummy", 666)
+        self.assertEqual(self.mat.cached, {"dummy": 666})
+        self.mat.adjustTD(0.5)
+        self.assertEqual(0.5, self.mat.theoreticalDensityFrac)
+        self.assertEqual(self.mat.cached, {})
+
+    def test_duplicate(self):
+        """Test the material duplication.
+
+        .. test:: Test the material base class.
+            :id: T_ARMI_MAT_PROPERTIES3
+            :tests: R_ARMI_MAT_PROPERTIES
+
+        .. test:: Materials shall calc mass fracs at init.
+            :id: T_ARMI_MAT_FRACS
+            :tests: R_ARMI_MAT_FRACS
+        """
+        mat = self.mat.duplicate()
+
+        self.assertEqual(len(mat.massFrac), len(self.mat.massFrac))
+        for key in self.mat.massFrac:
+            self.assertEqual(mat.massFrac[key], self.mat.massFrac[key])
+
+        self.assertEqual(mat.parent, self.mat.parent)
+        self.assertEqual(mat.refDens, self.mat.refDens)
+        self.assertEqual(mat.theoreticalDensityFrac, self.mat.theoreticalDensityFrac)
+
+    def test_cache(self):
+        """Test the material cache.
+
+        .. test:: Test the material base class.
+            :id: T_ARMI_MAT_PROPERTIES4
+            :tests: R_ARMI_MAT_PROPERTIES
```

Comment:
This doesn't feel relevant to the linked requirement. Let's chat offline.

Reply:
I agree, removed.